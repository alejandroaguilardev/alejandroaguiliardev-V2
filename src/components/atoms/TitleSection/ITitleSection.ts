export interface ITitleSection {
	title: string;
	description: string;
}
