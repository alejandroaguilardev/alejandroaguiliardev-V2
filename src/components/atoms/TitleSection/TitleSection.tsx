import { Box, Typography } from "@mui/material";
import { ITitleSection } from "./ITitleSection";

export const TitleSection = ({
	align = "center",
	title,
	description,
	descriptionTitle = [],
}: ITitleSection) => {
	return (
		<Box mb={4}>
			<Typography
				role="title"
				mb={1}
				variant="h2"
				align="center"
				sx={{
					fontSize: "2rem",
					fontWeight: "bold",
				}}
			>
				{title}
			</Typography>
			{description.map((text, index) => (
				<Typography
					role="description"
					align={align}
					key={text}
					mb={2}
					sx={{
						fontWeight: !!descriptionTitle[index] ? "bold" : "normal",
					}}
				>
					{text}
				</Typography>
			))}
		</Box>
	);
};
