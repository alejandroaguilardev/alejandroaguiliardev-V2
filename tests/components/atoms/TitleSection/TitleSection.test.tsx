import { getByRole, render, screen } from "@testing-library/react";
import { TitleSection } from "../../../../src/components/atoms/TitleSection/TitleSection";

describe("TitleSection", () => {
	const align = "right";
	const title = "title";
	const description = ["description 1 ", "Description 2"];
	const descriptionTitle = [true, false];

	test("toMacthSnapshot", () => {
		const { container } = render(
			<TitleSection title={title} description={description} />
		);

		expect(container).toMatchSnapshot();
	});

	test("Props Test", () => {
		render(
			<TitleSection
				align={align}
				title={title}
				description={description}
				descriptionTitle={descriptionTitle}
			/>
		);

		const h2 = screen.getByRole("title");
		const p = screen.getAllByRole("description");

		expect(h2.textContent).toContain(title);
		let i = 0;
		for (const element of p) {
			expect(element.textContent).toContain(description[i++]);
			expect(element.className).toContain(
				align[0].toUpperCase() + align.substring(1)
			);
		}
	});
});
