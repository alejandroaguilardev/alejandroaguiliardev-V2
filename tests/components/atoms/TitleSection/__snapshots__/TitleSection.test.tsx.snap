// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleSection toMacthSnapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1yjvs5a"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-162zxzj-MuiTypography-root"
      role="title"
    >
      title
    </h2>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1mqboxc-MuiTypography-root"
      role="description"
    >
      description 1 
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1mqboxc-MuiTypography-root"
      role="description"
    >
      Description 2
    </p>
  </div>
</div>
`;
